### PR TITLE
Python SDK enhancements

### PR DESCRIPTION
**Pull request by woop**
* Refactor CLI
* Support additional types with tests
* Improve YAML loading
* Improve List type performance
* Fix bug with source handling
* Improve exception handling

### CODE REVIEW

File: sdk/python/feast/client.py [hunk 1]
```diff
@@ -348,7 +367,7 @@ def field_values_to_df(
                         value_attr[feature_id]
                     ]
```
Comment (thirteen37) [inline sdk/python/feast/client.py]: If `FEAST_VALUE_ATTR_TO_DTYPE` had a mapping of `None` to `None`, this if statement is not necessary.

File: sdk/python/feast/client.py [hunk 2]
```diff
@@ -223,43 +234,39 @@ def _apply_feature_set(self, feature_set: FeatureSet):
...
-            raise Exception(
-                "Error while trying to apply feature set " + feature_set.name
-            )
+            applied_fs = FeatureSet.from_proto(apply_fs_response.feature_set)
+            feature_set._update_from_feature_set(applied_fs, is_dirty=False)
 
-        applied_fs = FeatureSet.from_proto(apply_fs_response.feature_set)
-        feature_set._update_from_feature_set(applied_fs, is_dirty=False)
+        except Exception as e:
+            print(e)
```
[the remark this answers is not in the document]
Comment (woop) [reply sdk/python/feast/client.py]: The whole SDK needs a clean up in terms of exception handling, validation, and logging. Would it be better if I added TODO comments?

[the remark this answers is not in the document]
Comment (woop) [reply sdk/python/feast/client.py]: Also unit tests

File: sdk/python/feast/client.py [hunk 3]
```diff
@@ -223,43 +234,39 @@ def _apply_feature_set(self, feature_set: FeatureSet):
...
 
         Args:
             feature_ids: List of feature ids in the following format
                          "feature_set_name:version:feature_name".
 
-            entity_data: Pandas Dataframe representing the requested features.
-
-                         There must be a column named "datetime" with dtype
-                         "datetime64[ns]" or "datetime64[ns],UTC".
+            entity_rwaleows: List of GetFeaturesRequest.EntityRow where each row
```
Comment (woop) [inline sdk/python/feast/client.py]: Flagging that typo

File: sdk/python/feast/client.py [hunk 4]
```diff
@@ -185,32 +186,42 @@ def list_feature_sets(self) -> List[FeatureSet]:
             feature_sets.append(feature_set)
         return feature_sets
 
-    def get_feature_set(self, name: str, version: int) -> FeatureSet:
+    def get_feature_set(self, name: str, version=None) -> FeatureSet:
```
[the remark this answers is not in the document]
Comment (woop) [reply sdk/python/feast/client.py]: Nice catch. I think I started with having the option of "latest" or an integer being provided, but decided that omitting the value would represent latest. Forgot to add the type back in.

File: sdk/python/feast/client.py (hunk 1 above)
Comment (woop) [reply sdk/python/feast/client.py]: Yip. The whole type_map.py file needs to be cleaned up actually. A lot of debt in there. Still a pending task.

File: sdk/python/feast/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (woop) [reply sdk/python/feast/client.py]: Cleaned these up for the client.py, still need to deal with the rest of the SDK. I've added cleaner formatting as well.

File: sdk/python/feast/client.py (hunk 1 above)
Comment (woop) [reply sdk/python/feast/client.py]: I removed this whole block of code. It was dead anyways. Problem solved!

**Pull request merged**